### DISCUSSION TITLE
Guidelines for creating and sharing examples?

### DISCUSSION THREAD

**Discussion by jzavala-gonzalez** [duckdb/duckdb-wasm#787] [Q&A] [answered]
Hey all! I was looking for duckdb-wasm examples and found the duckdb-wasm-examples organization which has some great repos with minimal code (including one for SvelteKit!). I'm thinking of forking some of them to create some more examples and I'd appreciate your thoughts on some questions before writing anything:

1) Is using Typescript for all examples strongly encouraged? I'm guessing that also including non-typed Javascript examples could lower the barrier to entry. (Example: The sveltekit-typescript repo)

2) Any simple datasets that benefit from duckdb? Maybe a scaled-down version of the taxi dataset in a single-file parquet and/or a csv? The idea is aiming towards something like iris or mtcars that is very tutorial-friendly 😃 

3) Any guidelines for sharing any new example repos? Some examples are posted within discussions (e.g. the bare-browser, esbuild-browser, etc.) while others are within the duckdb-wasm-examples org. It’s a bit tough to find simple example repos right now so any advice on how to make these more accesible is appreciated. I think the two aforementioned sources could be mentioned in the typescript API repo’s README so more people can find them. 

Overall, trying to get more familiar with using duckdb through the browser through examples. Thank you for any help you can provide!

**Comment by ankoh** [comment, marked as answer]
Hey @jmzg1229 , this is a great idea!
We haven't invested too much time yet into structured and forkable DuckDB-Wasm scaffolds.
duckdb-wasm-examples/ was created for that purpose, but we have yet to fill it with meaningful examples.
Any help here is very welcome, diversity among the examples would only push this forward.

Regarding your questions:

1. Yes, I only went with Typescript since that's what we used for the web shell. I agree that a simpler javascript version without transpiler would be very helpful.
2. I also agree that having a more structured example theme would help a lot. I used TPCH at scale factor 0.1 and 0.01 so far but those datasets are not really driving too much excitement. What's a demo dataset that you like to play around with?
3. If you want to contribute examples, I'd happily move them over to the duckdb-wasm-examples org (and add you there if you're interested). Whats missing for them still are probably rudimentary GitHub actions with dependabot updates, otherwise they get stale very quickly. As soon as we have that, we should totally include them in the docs and the READMEs.

Best,
Andre

**Reply by jzavala-gonzalez** [reply]
Thank you! I'll see what I code up and let you know

**Reply by jzavala-gonzalez** [reply]
Hi, I created an example repository using Vite and no front-end framework at [jzavala-gonzalez/duckdbwasm-vitebrowser](https://github.com/jzavala-gonzalez/duckdbwasm-vitebrowser). It runs a Javascript file based off one of the Typescript examples that you shared in a [previous discussion](https://github.com/duckdb/duckdb-wasm/discussions/375#discussioncomment-1677031). Feel free to give any feedback and I'll see how I can improve it. It's also open to pull requests in case you have anything specific in mind. Also, I haven't used GitHub actions and dependabot yet so if you know of any articles or good resources to read up on how to use it for these example repositories, send them over and I'll take a look.

**Reply by ankoh** [reply]
Great! Do you want to move it over to duckdb-wasm-examples?

**Reply by jzavala-gonzalez** [reply]
Just transferred it. Feel free to open issues/discussions on the repo there.
